File: backend/src/009-sistema_operativo/arch/arch.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import sys
import platform

print(platform.architecture())

# Obtener la arquitectura en bits (32/64)
def bits_arch():  # Esta función es más segura
    if sys.maxsize > 2**32:
        return 64
    else:
        return 32

File: backend/src/009-sistema_operativo/arch/arch.go
package main

import (
   "fmt"
   "runtime"
)

func BitsArch() int {
    return 32 << uintptr(^uintptr(0)>>63)
}

func main() {
    // Obtener la arquitectura
    fmt.Printf("%s\n", runtime.GOARCH)  // amd64

    // Obtener la arquitectura en bits
    fmt.Printf("%d\n", BitsArch())      // 64
}
